frontend/src/metabase/actions/hooks/use-action-form/utils.ts

```diff
@@ -116,39 +115,27 @@ export const getInputType = (param: Parameter, field?: Field) => {
   return "string";
 };
 
-export const generateFieldSettingsFromParameters = (
-  params: Parameter[],
-  fields?: Field[],
```

Comment:
Important context for the use of this function: it's only called in the case of implicit actions. At least that's my assumption, because this is the only place this function is called:
https://github.com/metabase/metabase/blob/3e45f4acb138aa954749bf25aac18471f0ad2ab0/frontend/src/metabase/actions/hooks/use-action-form/use-action-form.ts#L25-L30

Custom (query) actions are guaranteed to have truthy `visualization_settings?.fields`, but for implicit actions `visualization_settings?.fields` is always undefined.

Reply:
yeah, good catch, this wasn't always the case, but that code path is dead now. We used to populate the whole model on the actions payload, so we could pull all the field metadata and create richer forms from it.